### ISSUE TITLE
Allow passing batch_size via CLI

### ISSUE THREAD

**Issue by Muennighoff** [embeddings-benchmark/mteb#1025]
I think it'd make sense to allow passing the `batch_size`kwarg when using the cli argument `mteb run ...`? Some missing results are due to OOMs (https://github.com/embeddings-benchmark/mteb/pull/1014)

**Comment by isaac-chung** [comment]
Yeah. And within the tasks / evaluators, `batch_size` could be passed along as a kwarg?

**Comment by KennethEnevoldsen** [comment]
That has been the approach so far. I would actually probably prefer having encode_kwargs as an argument for the MTEB(model, encode_kwargs={"batch_size": 16}). Batch size seem very specific. This also allow passing e.g. length normalization.

**Comment by Muennighoff** [comment]
>  Yeah. And within the tasks / evaluators, batch_size could be passed along as a kwarg?

As @KennethEnevoldsen said you can already pass it in evaluation.run like e.g. here: https://github.com/ContextualAI/gritlm/blob/0cc9aeab83b90f2e22bcdd2b084d51507c624d95/evaluation/eval_mteb.py#L1206

Maybe it would help having that in the docs / one of the README examples so people know 🤔

**Comment by KennethEnevoldsen** [comment]
I am more than happy to implement the encode_kwargs if people think it is a good idea.
